v1/contracts/ResultComputation.sol

```diff
@@ -0,0 +1,52 @@
+pragma solidity ^0.5.0;
+
+import "./FixedPoint.sol";
+
+
+/**
+ * @title Computes vote results.
+ * @dev The result is the mode of the added votes, if the mode's frequency is >50%. Otherwise, the vote is unresolved.
+ */
+library ResultComputation {
+
+    using FixedPoint for FixedPoint.Unsigned;
+
+    struct ResultComputationData {
+        // Maps price to number of tokens that voted for that price.
+        mapping(int => FixedPoint.Unsigned) voteFrequency;
+        // The total votes that have been added.
+        FixedPoint.Unsigned totalVotes;
+        // The price that is the current mode, i.e., the price with the highest frequency in `voteFrequency`.
+        int currentMode;
+    }
+
+    /**
+     * @dev Returns whether the result is resolved, and if so, what value it resolved to. `price` should be ignored if
+     * `isResolved` is false.
+     */
+    function getResolvedPrice(ResultComputationData storage data) internal view returns (bool isResolved, int price) {
```

Comment:
As discussed IRL, it might be nice to have a parameter passed in for the GAT threshold in terms of number of votes, so we can check that as well.